components/brave_ads/browser/notification_helper_linux.cc

```diff
@@ -41,4 +72,52 @@ NotificationHelper* NotificationHelper::GetInstanceImpl() {
   return NotificationHelperLinux::GetInstanceImpl();
 }
 
+///////////////////////////////////////////////////////////////////////////////
+
+bool NotificationHelperWin::IsNotificationsEnabled() {
+  dbus::Bus::Options bus_options;
+  bus_options.bus_type = dbus::Bus::SESSION;
+  bus_options.connection_type = dbus::Bus::PRIVATE;
+  bus_options.dbus_task_runner = dbus_thread_linux::GetTaskRunner();
+
+  scoped_refptr<dbus::Bus> bus = base::MakeRefCounted<dbus::Bus>(bus_options);
+
+  dbus::ObjectProxy* dbus_proxy =
+      bus->GetObjectProxy(DBUS_SERVICE_DBUS, dbus::ObjectPath(DBUS_PATH_DBUS));
+
+  dbus::MethodCall name_has_owner_call(DBUS_INTERFACE_DBUS,
+                                       kMethodNameHasOwner);
+  dbus::MessageWriter writer(&name_has_owner_call);
+
+  writer.AppendString(kFreedesktopNotificationsName);
+
+  std::unique_ptr<dbus::Response> name_has_owner_response =
+      dbus_proxy->CallMethodAndBlock(&name_has_owner_call,
+          dbus::ObjectProxy::TIMEOUT_USE_DEFAULT);
+  dbus::MessageReader reader(name_has_owner_response.get());
+
+  bool owned = false;
+  if (name_has_owner_response && reader.PopBool(&owned) && owned) {
+    return true;
+  }
+
+  // If the service currently isn't running, maybe it is activatable
+  dbus::MethodCall list_activatable_names_call(DBUS_INTERFACE_DBUS,
+                                               kMethodListActivatableNames);
+
+  std::unique_ptr<dbus::Response> list_activatable_names_response =
+      dbus_proxy->CallMethodAndBlock(&list_activatable_names_call,
+          dbus::ObjectProxy::TIMEOUT_USE_DEFAULT);
+
+  if (list_activatable_names_response) {
```

Comment:
nit: I would do `!list_activatable_names_response` and return false

Reply:
Changed